multiqc/modules/mirtrace/mirtrace.py

```diff
@@ -0,0 +1,381 @@
+                    body[row[0]]=row[1:len(row)]
+        except ValueError as e:
+            raise e
+
+        if header[0] == 'CLADE':
+            for record in header[1:len(header)]:
+                s_name = self.clean_s_name(record, f['root'])
+                parsed_data = {}
+                idx = header[1:len(header)].index(record)
+                for clade in body:
+                    parsed_data[clade] = int(body[clade][idx])
+                if s_name in self.contamination_data:
+                    s_name = s_name + '_' + ''.join(random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits) for _ in range(6))
+                self.add_data_source(f, s_name)
+                self.contamination_data[s_name] = parsed_data
+        else:
+            log.debug('No valid data for contamination check')
+            return None
+
+
+    # Parse a miRTrace mirtrace-stats-mirna-complexity.tsv file
+    def parse_complexity(self, f):
+        try:
+            with open(f['fn'], 'rb') as file:
+                reader = csv.reader(file, delimiter="\t")
+                header = reader.next()
+                body = {}
+                for row in reader:
+                    body[row[0]]=row[1:len(row)]
+        except ValueError as e:
+            raise e
+
+        if header[0] == 'DISTINCT_MIRNA_HAIRPINS_ACCUMULATED_COUNT':
+            for record in header[1:len(header)]:
+                s_name = self.clean_s_name(record, f['root'])
+                parsed_data = {}
+                idx = header[1:len(header)].index(record)
+                for depth in body:
+                    parsed_data[depth] = int(body[depth][idx]) if body[depth][idx] else 0
+                if s_name in self.complexity_data:
+                    s_name = s_name + '_' + ''.join(random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits) for _ in range(6))
+                self.add_data_source(f, s_name)
+                self.complexity_data[s_name] = parsed_data
+        else:
+            log.debug('No valid data for miRNA complexity')
+            return None
+
+    # miRTrace QC Plot
+    def mirtrace_qc_plot(self):
+        """ Generate the miRTrace QC Plot"""
+
+        # Specify the order of the different possible categories
+        keys = OrderedDict()
+        keys['adapter_removed_length_ok']   =   { 'color': '#006837', 'name': 'Reads ≥ 18 nt after adapter removal' }
+        keys['adapter_not_detected']        =   { 'color': '#66bd63', 'name': 'Reads without adapter' }
+        keys['length_shorter_than_18']      =   { 'color': '#fdae61', 'name': 'Reads < 18 nt after adapter removal' }
+        keys['low_complexity']              =   { 'color': '#d73027', 'name': 'Reads with low complexity' }
+        keys['low_phred']                   =   { 'color': '#a50026', 'name': 'Reads with low PHRED score' }
+
+        # Construct a data structure for the plot - duplicate the samples for read 1 and read 2
+        data = {}
```

Comment:
There's no need to create this data dict here as the structure and keys are the same as `self.summary_data`. MultiQC will ignore any data keys not specified in `keys`, so you can just call the function with the raw data directly here:

```python
return bargraph.plot(self.summary_data, keys, config)
```